ui/src/wizard/AzureWizard.ts

```diff
@@ -80,10 +83,16 @@ export class AzureWizard<T extends IInternalActionContext> implements types.Azur
                         }
                     });
 
+                    const loadingQuickPick = this._showLoadingPrompt ? createQuickPick(this._context, {
+                        loadingPlaceHolder: 'Loading...'
+                    }) : undefined;
                     try {
                         this.currentStepId = getEffectiveStepId(step);
+                        loadingQuickPick?.show();
                         await step.prompt(this._context);
```

Comment:
I would suggest you create a test wizard (like for manual testing, not unit testing) with fake delays in basically any place you can possibly think of. I suspect your current implementation does not show the Loading quick pick in as many places as it should. For example, if I have prompt code like this:
```
public async prompt(context: IActionContext): Promise<void> {
    await delay(5000);
    await context.ui.showQuickPick(...);
    await delay(5000);
}
```
I don't think the loading quick pick will be shown during the second delay. Also `getSubWizard` can return a promise, so there's potential for delay there as well

Reply:
Unless we add a way to detect exactly when the UI inside of `step.prompt` is shown/hidden then I don't think handling these cases is possible only from the AzureWizard (and without changing the wizard step implementations).

An idea I had was to pass a function `runWithLoadingPrompt` to the step. And then the step can use this however it needs to. Instead of trying to know when the UI is shown/hidden.

Reply:
Can you use `ui.onDidFinishPrompt` somehow? We're already using it a few lines up for other reasons.

Reply:
Looks like that will work perfectly. Thanks!

Reply:
Ok, here is what it looks like with a `delay(3000)` after the prompt. I'm not sure how I feel about the step # in the title not incrementing right away. Thoughts?

https://user-images.githubusercontent.com/12476526/129802447-4687b8c6-0ad5-40e2-9fc9-46269fd8bec4.mov

Reply:
Meh probably not worth the effort to mess with the step number